host-controller/src/main/java/org/jboss/as/domain/controller/transformers/ServerGroupTransformers.java

```diff
@@ -47,15 +47,22 @@ static ChainedTransformationDescriptionBuilder buildTransformerChain() {
         //////////////////////////////////
         //The EAP/AS 7.x chains
 
-        // module-options was introduced in 13, WildFly 20
-        ResourceTransformationDescriptionBuilder currentTo13 = createBuilderFromCurrent(chainedBuilder, KernelAPIVersion.VERSION_13_0);
-        JvmTransformers.registerTransformers13_AndBelow(currentTo13);
+        // graceful-startup attribute was introduced in Kernel 16 (Wildfly 23)
+        ResourceTransformationDescriptionBuilder currentTo15 = createBuilderFromCurrent(chainedBuilder, KernelAPIVersion.VERSION_15_0)
+                .getAttributeBuilder()
+                .setDiscard(DiscardAttributeChecker.DEFAULT_VALUE, ServerGroupResourceDefinition.GRACEFUL_STARTUP)
+                .addRejectCheck(RejectAttributeChecker.DEFINED, ServerGroupResourceDefinition.GRACEFUL_STARTUP)
+                .end();
+
+        // module-options was introduced in Kernel 13 (WildFly Core 12, WildFly 20)
```

Comment:
// module-options was introduced in Kernel API 14 (WildFly Core 13, WildFly 21)

Reply:
Updated